File: backend/models/comment.py
from sqlalchemy import Column, String, Text, Integer

from .base import BaseModel


class Comment(BaseModel):
    """批注模型"""
    __tablename__ = "comments"

    text = Column(Text, nullable=False)
    target_text = Column(Text, nullable=False)  # 被批注的文本
    start_index = Column(Integer, nullable=False)
    end_index = Column(Integer, nullable=False)
    author = Column(String, nullable=False)
    author_id = Column(String, nullable=False)
    session_id = Column(String, nullable=False)
    timestamp = Column(Integer, nullable=False)  # 毫秒时间戳

File: frontend/src/types/comment.ts
export interface CommentReply {
  id: string;
  text: string;
  author: string;
  timestamp: number;
}

export interface Comment {
  id: string;
  text: string;
  targetText: string;
  startIndex: number;
  endIndex: number;
  author: string;
  timestamp: number;
  replies: CommentReply[];
}
